crates/bevy_audio/src/audio.rs

```diff
@@ -51,7 +51,7 @@ pub enum PlaybackMode {
 /// [`AudioSink`][crate::AudioSink] or [`SpatialAudioSink`][crate::SpatialAudioSink]
 /// components. Changes to this component will *not* be applied to already-playing audio.
 #[derive(Component, Clone, Copy, Debug, Reflect)]
-#[reflect(Default, Component, Debug)]
+#[reflect(Default, ComponentMut, Component, Debug)]
```

Comment:
Haven't done a review yet or a global "is this worth it" analysis, but at first glance needing to reflect `Component` _and_ `ComponentMut` on every component feels like a painful UX price to pay.

Reply:
Yeah, I think that everyone (including the author) agrees that this is the biggest problem with the current PR.

Reply:
Agreed. The issues I encountered that led me here are:

1. The implementation for `ReflectComponent` included mutable methods (e.g., `reflect_mut`) which fail to compile in this PR, so they have to removed.
2. Ideally, we'd have `ReflectComponent` and `ReflectImmutableComponent`, where `ReflectComponent` requires `ComponentMut` and `ReflectImmutableComponent` only requires `Component`.
3. Reflection support for immutable components is a must, since we'd potentially like to use it for types like `Name`, `Parent`, `Children`, etc., which are all _very_ important.

I'll experiment with the PR some more to see if I can remove the need to `#[reflect(ComponentMut)]`, because that's definitely the worst part of this PR right now.

Reply:
Would making this information dynamic as metadata of the component help? All of these problems seem to be coming about from information solely existing in the type system.

Reply:
I believe I have addressed the reflection issue entirely now, as well as dynamic components, _and_ can offer a reliable and safe `ComponentImmutable` trait that acts as `!ComponentMut`.

Reply:
> Would making this information dynamic as metadata of the component help? All of these problems seem to be coming about from information solely existing in the type system.

The information not being available in the type system is an absolute non-starter for me. Letting users write `Query<&mut MyImmutableFoo>` only for it to crash at runtime massively degrades the developer experience.

Reply:
Agreed. This information needs to exist in both the type-system and dynamically.

Reply:
Read that comment again.

> **solely** existing in the type system